daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/DecodeV1.scala

```diff
@@ -51,14 +51,18 @@ private[archive] class DecodeV1(minor: LV.Minor) extends Decode.OfPackage[PLF.Pa
         None
       }
 
-    val env = DecoderEnv(
+    val env0 = DecoderEnv(
       packageId,
       internedStrings,
       internedDottedNames,
+      IndexedSeq(),
       Some(dependencyTracker),
       None,
       onlySerializableDataDefs,
     )
+    val internedTypes = decodeInternedTypes(env0, lfPackage.getInternedTypesList.asScala.toSeq)
```

Comment:
@remyhaemmerle-da @S11001001 Is `.asScala.toSeq` a bad idea for performance?

Reply:
The result of `asScala` is a `collection.Seq`. (It is not a `collection.immutable.Seq`.) If it compiles without `.toSeq`, that is cheaper (constant in fact), as `toSeq` returns the latter.